### PR TITLE
Submit to FMS

### PR DESCRIPTION
**Pull request by a8wright**
In this pull request I have added several features:

* Made the logging more robust (with color)
* Made it so that there is a config file that can be changed to set the environment variables and other parameters in the future. 
* Pulled generators into their own folder
* Code for submitting VCF files to the FMS. 

Future:
* Add other file types to be submitted to FMS
* Create a pipeline in GOCD

@mgrbyte I was able to get app.py to find the modules. Not quite what you were suggesting and we should probably change this in a future refactor but for now this is simple and works.

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -1,40 +1,64 @@
-import logging
+import logging, coloredlogs, yaml, os, sys, urllib3, requests
```
Comment (mgrbyte) [inline src/app.py]: non-functional minor things/points:
Although Python aloughs comma separated imports, they're not often used in the wild - 
Python's own coding standard (that a lot of folks just adhere to) recommends one import statement per line.

Its also generally good practice/standard Python coding convention (might even be in PEP8) to organise imports like:
- builtin modules
<newline separator>
- third-party modules
<newline separator>
- local project modules

Makes it easy to at a glance where the module is from without searching.

File: src/app.py [hunk 2]
```diff
@@ -1,40 +1,64 @@
...
 import os
-
 import click
 
-from vcf_file_generator import VcfFileGenerator
-from orthology_file_generator import OrthologyFileGenerator
-from daf_file_generator import DafFileGenerator
-from expression_file_generator import ExpressionFileGenerator
+from generators import *
```
Comment (mgrbyte) [inline src/app.py]: Also a no-no is the use of `import *` - best to explicitly name the variables you want to bring in, or alias the module if you want brevity (e.g: `import generators as g`)

File: src/generators/vcf_file_generator.py [hunk 3]
```diff
@@ -4,9 +4,10 @@
 import logging
 import os
 import time
-
 import pyfaidx
 
+from upload import *
```
Comment (mgrbyte) [inline src/generators/vcf_file_generator.py]: same as above about `import *`

File: src/generators/vcf_file_generator.py [hunk 4]
```diff
@@ -101,7 +102,7 @@ def _consume_data_source(self):
         assembly_chr_variants = defaultdict(lambda : defaultdict(list))
         assembly_species = {}
         for variant in self.variants:
-            assembly = variant['assembly']
+            assembly = 'R6' if variant['assembly'].startswith('R6') else variant['assembly'].replace('.', '').replace('_', '')
```
Comment (mgrbyte) [inline src/generators/vcf_file_generator.py]: Nit-pick: personally prefer and regular if-else block over these inline conditionals for readability unless very short/no work going on inside the inline conditional.

File: src/upload/upload.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    with open(os.path.join(save_path,filename), 'rb') as f:
+        for chunk in iter(lambda: f.read(4096), b''):
+            hash_md5.update(chunk)
+    logger.info('{}: Finished generating md5 hash: {}'.format(worker, hash_md5.hexdigest()))
+
+    return hash_md5.hexdigest()
+
+
+def upload_file(worker, filename, save_path, upload_file_prefix, config_info):
+    file_to_upload = {upload_file_prefix: open(os.path.join(save_path, filename), 'rb')}
```
Comment (mgrbyte) [inline src/upload/upload.py]: Probably better to have this ensure the file is closed, e.g:
```python
with open(....) as fp:
    ... rest of function as is now using fp instead ...
````

File: src/generators/vcf_file_generator.py (hunk 4 above)
Comment (a8wright) [reply src/generators/vcf_file_generator.py]: Thanks for the point. I will leave it in for now. The R6 thing will be fixed in the data sometime soon so I will leave it in for now.

**Comment by a8wright** [comment]
@mgrbyte I have made changes correspond with your requests (minus one).

File: src/app.py [hunk 6]
```diff
@@ -1,8 +1,16 @@
-import logging, coloredlogs, yaml, os, sys, urllib3, requests
+import sys
```
Comment (mgrbyte) [inline src/app.py]: think you're missing `import logging`

**Pull request merged**